src/signals/settings/roles/containers/RoleFormContainer/index.js

```diff
@@ -32,37 +35,55 @@ export const RoleFormContainer = ({
   },
   onPatchRole,
   onSaveRole,
+  showGlobalNotification,
+  onResetResponse,
   userCan,
 }) => {
   const { roleId } = useParams();
+  const location = useLocation();
+  const history = useHistory();
   const role = list.find(item => item.id === roleId * 1);
   const title = `Rol ${roleId ? 'wijzigen' : 'toevoegen'}`;
+  const redirectURL = {
+    pathname: location.referrer || routes.roles,
+  };
+
+  useEffect(() => {
+    let message;
+
+    if (responseSuccess) {
+      message = roleId ? 'Gegevens opgeslagen' : 'Rol toegevoegd';
+    };
+    onResetResponse();
+
+    if (!message) return;
+
+    showGlobalNotification({
+      variant: VARIANT_SUCCESS,
+      title: message,
+      type: TYPE_LOCAL,
+    });
+
+    if (responseSuccess) {
+      history.push(redirectURL.pathname, redirectURL.state);
```

Comment:
`state` Isn't used by the page you're pushing an item on the stack for, please remove from this function as parameter.

Reply:
gedaan